patches/server/0906-Add-BrainMemoryUpdateEvent.patch

```diff
@@ -0,0 +1,50 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Togira <70365614+Togira123@users.noreply.github.com>
+Date: Mon, 23 May 2022 20:12:27 +0200
+Subject: [PATCH] Add BrainMemoryUpdateEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/ai/Brain.java b/src/main/java/net/minecraft/world/entity/ai/Brain.java
+index 91d266bd1fc37c9e284a99bdbb465d0e3100305a..8e0024694c8fe76f44c9211c050522dc65827cf0 100644
+--- a/src/main/java/net/minecraft/world/entity/ai/Brain.java
++++ b/src/main/java/net/minecraft/world/entity/ai/Brain.java
+@@ -163,7 +163,38 @@ public class Brain<E extends LivingEntity> {
+             if (memory.isPresent() && this.isEmptyCollection(memory.get().getValue())) {
+                 this.eraseMemory(type);
+             } else {
+-                this.memories.put(type, memory);
++                // Paper start - check whether memory is being added/changed/deleted
```

Comment:
Clean this up by only putting the change type into a var since the rest of the call is always the same